File: 프로그래머스/unrated/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.py
def solution(s):
    answer = [-1] * len(s)
    
    for i in range(1, len(s)):
        for j in range(1, i+1):       # 앞에 나온 글자들만 가까운 순부터 조사
            if s[i] == s[i-j]:      # 같은 글자가 있다면 거리 반영
                answer[i] = j
                break        
        
    return answer

File: 프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.java
import java.util.*;

class Solution {
    public int[] solution(String s) {
        int[] answer = new int[s.length()];
        char[] characters = s.toCharArray();
        int[] latest = new int[26];     // 알파벳별 최신 인덱스 모음
        
        for (int i = 0; i < 26; i++) {
            latest[i] = -1;
        }
        
        for (int i = 0; i < s.length(); i++) {
            int idx = characters[i] - 'a';      // 해당 알파벳 인덱스 확인
            if (latest[idx] > -1) {
                answer[i] = i - latest[idx];            // 앞에 같은 글자 있을 경우             
            } else {
                answer[i] = -1;
            }
            latest[idx] = i;                    // 최신 인덱스 갱신
        }
        
        
        return answer;
    }
}